### PR TITLE
Enable android builds via vcpkg

### PR DESCRIPTION
**Pull request by m-kuhn**
There is plenty of cleanup in this PR. And a major refactoring of the build process. Too much to list here.

So while there is a lot of GOOD things here, I'll focus on the ugly:

- downgrade of proj (well, that's fixable ...)
- we are currently working on a fork of vcpkg that adds support for autotools (autoconf, make etc) that would need to be upstreamed
- this switches the build to static libs, with the consequence that only providers that are implemented for static builds are usable (like on iOS).

Available:
- gdal
- ogr
- postgresql
- wms
- wcs
- delimtedtext
- afs
- ams
- spatialite
- wfs
- virtual

Not available:
- mdal
- gpx
- db2
- mssql
- ~~geonode~~ (GUI-only, not relevant to QField)

_Note that some of these may be available through gdal_

TODO:

- [x] Create packages (androiddeployqt)
- [x] Make sure all platforms build
- [x] Check demo projects
- [x] Check proj.db
- [x] Min api version 21
- [x] Pass version, name etc to cmake build
- [x] Sign apk
- [x] Check SSL
- [x] Revive the beloved fairy 🧚 
- [x] Recover icon logic (nightly, pr etc)
- [x] package qgis assets like svg. It's present in cmake but doesn't end up in the apk.

### CODE REVIEW

**Comment by nirvn** [comment]
The providers thing seems to be the only real blocker to move master to this new build system. I've therefore worked out static build support for key providers here (https://github.com/qgis/QGIS/pull/45737). Most important one being WFS since we spent a lot of energy improving this in 1.9 and 1.10.

**Comment by nirvn** [comment]
@m-kuhn , you beat me to it :)

File: vcpkg/overlay/qgis/portfile.cmake [hunk 1]
```diff
@@ -1,7 +1,5 @@
-vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
-
-set(QGIS_REF final-3_22_0)
-set(QGIS_SHA512 01e84756b68a33e107f9d00083acb8dc3970aad944f09f034dfccc4983e80284b0c6d00547f853895446ac95e5c8b150677ee3253421395b83badcc2168fb032)
+set(QGIS_REF backport-45737-to-release-3_22)
+set(QGIS_SHA512 0)
```
Comment (nirvn) [inline vcpkg/overlay/qgis/portfile.cmake:2]:
```suggestion
set(QGIS_SHA512 4c6614fef7fde876178d49248e445363aeec359fbb291f75d467e1de45f402674168f313531e459465a69789526bc499e9871a9e42ee52922b80d3db0bd64072)
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for aee5f871fa643a9c15ad4045d6dcced244361369:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-8b04707d8d0d5aaf8eab2e918ed76dd5138ba074-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-8b04707d8d0d5aaf8eab2e918ed76dd5138ba074-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-8b04707d8d0d5aaf8eab2e918ed76dd5138ba074-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-8b04707d8d0d5aaf8eab2e918ed76dd5138ba074-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8c8d9b37a69638ce515675a07232ef892a3bad8c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-bd2f08dfc4cff2750b280830ba4603107acc933d-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-bd2f08dfc4cff2750b280830ba4603107acc933d-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-bd2f08dfc4cff2750b280830ba4603107acc933d-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-bd2f08dfc4cff2750b280830ba4603107acc933d-x86.apk)

**Comment by nirvn** [comment]
Build is working fine 🥳

Now onto what's missing: we need to build QGIS with spatialite on, ie that line:

list(APPEND QGIS_OPTIONS -DWITH_QSPATIALITE:BOOL=OFF)

**Comment by nirvn** [comment]
Building error:

```
FAILED: output/lib/sqldrivers/libqsqlspatialite_x86.so 
: && /usr/local/lib/android/sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=i686-none-linux-android21 --gcc-toolchain=/usr/local/lib/android/sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64 --sysroot=/usr/local/lib/android/sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -mstackrealign -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wreturn-type-c-linkage -Woverloaded-virtual -Wimplicit-fallthrough -Qunused-arguments -O0 -fno-limit-debug-info  -Wl,--exclude-libs,libgcc.a -Wl,--exclude-libs,libgcc_real.a -Wl,--exclude-libs,libatomic.a -static-libstdc++ -Wl,--build-id -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments    -Qunused-arguments -shared -Wl,-soname,libqsqlspatialite_x86.so -o output/lib/sqldrivers/libqsqlspatialite_x86.so src/providers/spatialite/qspatialite/CMakeFiles/qsqlspatialite.dir/qsqlspatialite_autogen/mocs_compilation.cpp.o src/providers/spatialite/qspatialite/CMakeFiles/qsqlspatialite.dir/qsql_spatialite.cpp.o src/providers/spatialite/qspatialite/CMakeFiles/qsqlspatialite.dir/smain.cpp.o src/providers/spatialite/qspatialite/CMakeFiles/qsqlspatialite.dir/moc_qsql_spatialite.cpp.o src/providers/spatialite/qspatialite/CMakeFiles/qsqlspatialite.dir/moc_smain.cpp.o -L/home/runner/src/buildtrees/qgis/x86-android-dbg/src/core   -L/home/runner/src/buildtrees/qgis/x86-android-dbg/src/gui /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libsqlite3.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libspatialite.a  src/core/libqgis_core.a  src/providers/wms/libprovider_wms_a.a  src/providers/wcs/libprovider_wcs_a.a  src/providers/delimitedtext/libprovider_delimitedtext_a.a  src/providers/arcgisrest/libprovider_arcgisfeatureserver_a.a  src/providers/arcgisrest/libprovider_arcgismapserver_a.a  src/providers/spatialite/libprovider_spatialite_a.a  src/providers/wfs/libprovider_wfs_a.a  src/providers/virtual/libprovider_virtuallayer_a.a  src/providers/postgres/libprovider_postgres_a.a  src/auth/basic/libauthmethod_basic_a.a  src/auth/esritoken/libauthmethod_esritoken_a.a  src/auth/identcert/libauthmethod_identcert_a.a  src/auth/pkipaths/libauthmethod_pkipaths_a.a  src/auth/pkipkcs12/libauthmethod_pkcs12_a.a  src/auth/oauth2/libauthmethod_oauth2_a.a  src/core/libqgis_core.a  src/providers/wms/libprovider_wms_a.a  src/providers/wcs/libprovider_wcs_a.a  src/providers/delimitedtext/libprovider_delimitedtext_a.a  src/providers/arcgisrest/libprovider_arcgisfeatureserver_a.a  src/providers/arcgisrest/libprovider_arcgismapserver_a.a  src/providers/spatialite/libprovider_spatialite_a.a  src/providers/wfs/libprovider_wfs_a.a  src/providers/virtual/libprovider_virtuallayer_a.a  src/providers/postgres/libprovider_postgres_a.a  src/auth/basic/libauthmethod_basic_a.a  src/auth/esritoken/libauthmethod_esritoken_a.a  src/auth/identcert/libauthmethod_identcert_a.a  src/auth/pkipaths/libauthmethod_pkipaths_a.a  src/auth/pkipkcs12/libauthmethod_pkcs12_a.a  src/auth/oauth2/libauthmethod_oauth2_a.a  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Sql_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Xml_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Svg_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Network_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Concurrent_x86.so  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libqt5keychaind.a  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5AndroidExtras_x86.so  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libproj.a  /home/runner/builddir/vcpkg_installed/x86-android/lib/libgdal.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libcurl-d.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libexpat.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libgeos_cd.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libgeosd.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libgif.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libjson-c.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libgeotiff.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libtiffd.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/liblzmad.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libjpeg.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libproj.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libpng16d.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libpq.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libwebpd.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libcpufeaturesd.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libxml2.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libopenjp2.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libproj.a  -lm  -ldl  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libsqlite3.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libtiffd.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libjpeg.a  -lm  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/liblzmad.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libz.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libzstdd.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libspatialindex.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libzip.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libprotobuf-lited.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libz.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libexiv2.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libiconv.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libcharset.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libexiv2-xmp.a  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5PrintSupport_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Widgets_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Gui_x86.so  /home/runner/builddir/vcpkg_installed/x86-android/lib/libzstd.a  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5SerialPort_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Positioning_x86.so  /home/runner/work/QField/Qt/5.14.2/android/lib/libQt5Core_x86.so  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libqca.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libexpat.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libsqlite3.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libspatialite.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libssl.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libcrypto.a  -pthread  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libpq.a  -ldl  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libpgport.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libpgcommon.a  /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libgeos_cd.a /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libgeosd.a /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/librttopo.a /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libfreexl.a /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libbz2d.a /home/runner/builddir/vcpkg_installed/x86-android/debug/lib/libgslcblasd.a && :
../.././../src/atialite-5-f1515a017e.clean/src/spatialite/spatialite.c:28194: error: undefined reference to '__android_log_print'

```

**Comment by nirvn** [comment]
@m-kuhn , for the record, non-spatialite providers, e.g. delimited text provider, works on Android. We're very close to being there :)

**Comment by nirvn** [comment]
Alright, we got spatialite, wfs et al in QField unlocked.

One thing, gdal says it isn't built with libspatialite support (?¡),  which in turns gets the wfs to throw this complaint:

Cannot create temporary SpatiaLite cache.: GDALCreate() failed: OGR was built without libspatialite support
... sorry, creating/writing any SpatiaLite DB is unsupported

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 35be161ec13820324a932d59970e278a447b9bec:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-be9bfa7c646fe4a5f3e537315cf9b565e5a3c8dc-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-be9bfa7c646fe4a5f3e537315cf9b565e5a3c8dc-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-be9bfa7c646fe4a5f3e537315cf9b565e5a3c8dc-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-be9bfa7c646fe4a5f3e537315cf9b565e5a3c8dc-x86.apk)

**Comment by m-kuhn** [comment]
Apparently that's an opt-in "feature" of the gdal port. Needs to be enabled through the vcpkg.json file.

**Comment by nirvn** [comment]
Relevant bit from config:

checking for spatialite... checking sqlite3.h usability... yes
checking sqlite3.h presence... yes
checking for sqlite3.h... yes
checking for spatialite.h in /usr/include or /usr/local/include... not found : spatialite support disabled

**Comment by nirvn** [comment]
Explicitly stating the include path via `with_spatialite="${CURRENT_INSTALLED_DIR}/include"` seems to get us in a better place but still not quite there yet.

Before, we'd get this error:
> checking for spatialite... disabled

Now, with the include path, we get:
> checking for spatialite... checking for spatialite_init in -lspatialite... no

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a0408ba251c2386bfe7c8bcf178bc2c1f25a26f0:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-b3a1cc131203ced580aefa1654fd95190f9e33a4-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-b3a1cc131203ced580aefa1654fd95190f9e33a4-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-b3a1cc131203ced580aefa1654fd95190f9e33a4-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-b3a1cc131203ced580aefa1654fd95190f9e33a4-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f469d121232f2f2a74c9fc2d2ee57dcd2c685ce8:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-45b5153e16d10cd0d991b3007b21c713707b374c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-45b5153e16d10cd0d991b3007b21c713707b374c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-45b5153e16d10cd0d991b3007b21c713707b374c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-45b5153e16d10cd0d991b3007b21c713707b374c-x86.apk)

**Comment by nirvn** [comment]
@m-kuhn , the latest round of APKs still has "QField" as app name. Were you aiming at resurrecting the "QField PR XXXX" / "QField Dev" names?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6035892289dda9580f410fcbd5f522a863eafec1:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-5ea3bcc9ed7a493bfafa5b935729497685b07f78-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-5ea3bcc9ed7a493bfafa5b935729497685b07f78-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-5ea3bcc9ed7a493bfafa5b935729497685b07f78-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-5ea3bcc9ed7a493bfafa5b935729497685b07f78-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for cc412307b8776673439802a2a7cbed63746e6804:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-993dbcaabca3b70c7cc06e6563dca8fbdba7d1ed-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-993dbcaabca3b70c7cc06e6563dca8fbdba7d1ed-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-993dbcaabca3b70c7cc06e6563dca8fbdba7d1ed-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-993dbcaabca3b70c7cc06e6563dca8fbdba7d1ed-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6e7c9ca71e26a9b773f6e3500a386e8e98ba9b7d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-ec8b5a6609e38523eca9bc739f5a2c2f78b725c2-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-ec8b5a6609e38523eca9bc739f5a2c2f78b725c2-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-ec8b5a6609e38523eca9bc739f5a2c2f78b725c2-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-ec8b5a6609e38523eca9bc739f5a2c2f78b725c2-x86.apk)

**Comment by nirvn** [comment]
This looks good to go for me.

**Review by nirvn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b66ce9f1198aba096cedac25bf2e566487a1bfad:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-588117f1534609f44ad450f91f12a38899853176-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-588117f1534609f44ad450f91f12a38899853176-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-588117f1534609f44ad450f91f12a38899853176-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2237-588117f1534609f44ad450f91f12a38899853176-x86.apk)

**Pull request merged**